pilot/pkg/model/config_test.go

```diff
@@ -177,42 +178,70 @@ func TestServiceKey(t *testing.T) {
 func TestSubsetKey(t *testing.T) {
 	hostname := model.Hostname("hostname")
 	cases := []struct {
+		got      string
 		hostname model.Hostname
 		subset   string
 		port     *model.Port
 		want     string
+		err      error
 	}{
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Name: "http", Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound|http|subset|hostname",
+			want:     "outbound.http.subset.hostname",
 		},
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound||subset|hostname",
+			want:     "outbound.80.subset.hostname",
 		},
 		{
 			hostname: "hostname",
 			subset:   "",
 			port:     &model.Port{Name: "http", Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound|http||hostname",
+			want:     "outbound.http._default_.hostname",
```

Comment:
Where all is this being generated ? You have to make sure you update all references to DestinationRules to always use non empty subsets.

Reply:
This is converted back to "" - I asked in separate comment to make it "_", it is parsed to empty.

Reply:
This is updated to "_"